CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKHitTester.java

```diff
@@ -356,96 +430,61 @@ public int OffscreenRender(List<CVKRenderable> hitTestRenderables) {
             inheritanceInfo.queryFlags(0);
             inheritanceInfo.pipelineStatistics(0);
-
-	    // TODO Is this needed or set in RecordHitTestCommandBuffer() ?
-            // Set the dynamic viewport and scissor
-            //commandBuffer.viewPortCmd(cvkSwapChain.GetWidth(), cvkSwapChain.GetHeight(), stack);
-            //commandBuffer.scissorCmd(cvkDevice.GetCurrentSurfaceExtent(), stack);
             
-            // Check flags and render the nodes and connections
-            /// Loop through command buffers of hit test objects and record their buffers
+            // Loop through command buffers of hit test objects and record their buffers
             hitTestRenderables.forEach(renderable -> {
-            // TODO HYDRA - WIP HIT TESTER
-//                if (renderable.GetVertexCount() > 0) {
-//                    renderable.RecordHitTestCommandBuffer(inheritanceInfo, 0);
-//                    vkCmdExecuteCommands(commandBuffer.GetVKCommandBuffer(), renderable.GetCommandBuffer(0));
-//                }
+                if (renderable.GetVertexCount() > 0) {
+                    renderable.RecordHitTestCommandBuffer(inheritanceInfo, 0);
+                    vkCmdExecuteCommands(commandBuffer.GetVKCommandBuffer(), renderable.GetHitTestCommandBuffer(0));
+                }
             });
             
-            vkCmdEndRenderPass(commandBuffer.GetVKCommandBuffer());
+            commandBuffer.EndRenderPass();
         
             // Pre Draw Barrier
-            commandBuffer.pipelineImageMemoryBarrierCmd(cvkImage.GetImageHandle(), 
-                    VK_IMAGE_LAYOUT_UNDEFINED, VK_IMAGE_LAYOUT_GENERAL,//VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL, VK_IMAGE_LAYOUT_GENERAL,    // Old/New Layout
-                    0, VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT,                                // Src/Dst Access mask
-                    VK_PIPELINE_STAGE_ALL_COMMANDS_BIT, VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT, //VK_PIPELINE_STAGE_ALL_COMMANDS_BIT, VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT,  // Src/Dst Stage mask
-                    0, 1);                                                                  // baseMipLevel/mipLevelCount
+            commandBuffer.PipelineImageMemoryBarrier(cvkImage.GetImageHandle(), 
+                    VK_IMAGE_LAYOUT_UNDEFINED, VK_IMAGE_LAYOUT_GENERAL,                                 // Old/New Layout
+                    0, VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT,                                            // Src/Dst Access mask
+                    VK_PIPELINE_STAGE_ALL_COMMANDS_BIT, VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT,  // Src/Dst Stage mask
+                    0, 1);                                                                              // baseMipLevel/mipLevelCount
             
+            cvkImage.SetLayout(VK_IMAGE_LAYOUT_GENERAL);
             // TODO Do we need this?
-            CVKDevice.WaitIdle();
+            //CVKDevice.WaitIdle();
```

Comment:
Can these comments be removed?

Reply:
Not sure yet, going to look into the freezing issue

Reply:
Actually I removed this, shouldnt need synchronisation between different render passes